src/lib.rs

```diff
@@ -47,39 +48,45 @@ impl PrivateFS {
     })
   }
 
-  pub fn include(&self, path: impl AsRef<Path>) -> Result<(), Box<dyn Error>> {
+  pub fn include(&self, source_path: impl AsRef<Path>, destination_path: Option<&str>) -> Result<(), Box<dyn Error>> {
     // Get our pathbuf to the file to include
-    let mut inner_path = path.as_ref().to_owned();
+    let mut inner_path = source_path.as_ref().to_owned();
 
     // If the path given is not absolute then it's relative to the dir we
     // ran the test from
     let is_relative = inner_path.is_relative();
     if is_relative {
-      inner_path = self.ran_from.join(&path);
+      inner_path = self.ran_from.join(&source_path);
     }
 
     // Get our working directory
     let dir = self.directory.path().to_owned();
 
-    // Make the relative path of the file in relation to our temp file
-    // system based on if it was absolute or not
-    let relative = if !is_relative {
-      inner_path
-        .components()
-        .filter(|c| *c != Component::RootDir)
-        .collect::<PathBuf>()
-    } else {
-      path.as_ref().into()
+    let destination_path = match destination_path {
+      None => {
+        // If the destination path is unspecified, we mount the file in the root directory
+        // of the test's private filesystem
+        // TODO: return meaningful error message
+        let filename = inner_path.file_name().unwrap().to_owned();
+        dir.join(filename)
+      },
+      Some(p) => {
+        let p = PathBuf::from(p);
+        if !p.is_relative() {
+          // TODO: do we want to panic here?
+          panic!("The destination path for included files must be a relative path. {:?} isn't.", p);
```

Comment:
I think we want to panic given this is going to be used in a test and panicing makes sense to indicate failure and I don't see assay libs being used outside testing.